substrate/frame/identity/src/lib.rs

```diff
@@ -959,6 +998,126 @@ pub mod pallet {
+
+			// `take` any storage items keyed by `target`
+			let id = <IdentityOf<T>>::take(&who).ok_or(Error::<T>::NotNamed)?;
+			let fields = id.info.additional.len() as u32;
+			let (subs_deposit, sub_ids) = <SubsOf<T>>::take(&who);
+			// check witness data
+			let actual_subs = sub_ids.len() as u32;
+			ensure!(actual_subs <= num_subs, Error::<T>::TooManySubAccounts);
+			for sub in sub_ids.iter() {
+				<SuperOf<T>>::remove(sub);
+			}
+
+			// unreserve any deposits
+			let deposit = id.total_deposit().saturating_add(subs_deposit);
+			let err_amount = T::Currency::unreserve(&who, deposit);
+			debug_assert!(err_amount.is_zero());
+
+			// Finally, call the handler.
+			T::ReapIdentityHandler::on_reap_identity(&who, fields, actual_subs)?;
+			Self::deposit_event(Event::IdentityReaped { who });
+			Ok(())
+		}
+
+		/// Update the deposits held by `target` for its identity info.
+		///
+		/// Parameters:
+		/// - `target`: The account for which to update deposits.
+		///
+		/// May be called by any signed origin.
+		#[pallet::call_index(16)]
+		#[pallet::weight(T::WeightInfo::poke_deposit())]
+		pub fn poke_deposit(origin: OriginFor<T>, target: AccountIdLookupOf<T>) -> DispatchResult {
+			// No locked check: used for migration.
+			// anyone or root (so that the system can call it for identity migration)
+			let _ = ensure_signed_or_root(origin)?;
+			let target = T::Lookup::lookup(target)?;
+
+			// Identity Deposit
+			ensure!(IdentityOf::<T>::contains_key(&target), Error::<T>::NoIdentity);
+			let new_id_deposit = IdentityOf::<T>::try_mutate(
+				&target,
+				|registration| -> Result<BalanceOf<T>, DispatchError> {
+					let reg = registration.as_mut().ok_or(Error::<T>::NoIdentity)?;
+					// Calculate what deposit should be
+					let field_deposit = T::FieldDeposit::get()
+						.saturating_mul(BalanceOf::<T>::from(reg.info.additional.len() as u32));
+					let new_id_deposit = T::BasicDeposit::get().saturating_add(field_deposit);
+
+					// Update account
+					Self::rejig_deposit(&target, reg.deposit, new_id_deposit)?;
+
+					reg.deposit = new_id_deposit;
+					Ok(new_id_deposit)
+				},
+			)?;
+
+			// Subs Deposit
+			let new_subs_deposit = SubsOf::<T>::try_mutate(
+				&target,
+				|(current_subs_deposit, subs_of)| -> Result<BalanceOf<T>, DispatchError> {
+					let new_subs_deposit = Self::subs_deposit(subs_of.len() as u32);
+					Self::rejig_deposit(&target, *current_subs_deposit, new_subs_deposit)?;
+					*current_subs_deposit = new_subs_deposit;
+					Ok(new_subs_deposit)
+				},
+			)?;
+
+			Self::deposit_event(Event::DepositUpdated {
+				who: target,
+				identity: new_id_deposit,
+				subs: new_subs_deposit,
+			});
+			Ok(())
+		}
+
+		/// Lock the pallet.
+		///
+		/// Origin must be the `LockerOrigin`.
+		#[pallet::call_index(17)]
+		#[pallet::weight(T::WeightInfo::lock_pallet())]
+		pub fn lock_pallet(origin: OriginFor<T>) -> DispatchResult {
+			T::LockerOrigin::ensure_origin(origin)?;
+			Locked::<T>::put(true);
```

Comment:
In order to be as explicit as possible with this, we can do `try_mutate` instead of `put` and return an error in case the flag was already `true`. Not a requirement, but more defensive. Same for the unlocking extrinsic.

Reply:
Returning an error and putting the same value are both no-ops though. What would it defend against?

Reply:
While the effects in storage would be the same, if you run `lock_pallet` with the flag already set, it's more obvious/alarming to the caller if they get an error that something's not right, because they'd call that only if they thought the pallet wasn't already locked. Functionally it makes no difference, the fees would be charged anyway. I'm fine with whatever you decide here.